### PR TITLE
Adding Boolean Tfhe-rust types and BitAnd op

### PR DESCRIPTION
**Pull request by WoutLegiest**
Adding the Boolean tfhe-rust types, a simple bitand operations and including an update of the tests

### CODE REVIEW

File: tests/tfhe_rust/ops.mlir [hunk 1]
```diff
@@ -3,7 +3,30 @@
...
+!bsks = !tfhe_rust.bool_server_key
+
 module {
+  // CHECK-LABEL: func @test_create_trivial_bool
+  func.func @test_create_trivial_bool(%bsks : !bsks) {
+    %0 = arith.constant 1 : i1
+    %1 = arith.constant 1 : i1
+
+    %e1 = tfhe_rust.create_bool_trivial %bsks, %0 : (!bsks, i1) -> !tfhe_rust.eb
+    %e2 = tfhe_rust.create_bool_trivial %bsks, %1 : (!bsks, i1) -> !tfhe_rust.eb
```
Comment (j2kun) [inline tests/tfhe_rust/ops.mlir:15]:
```suggestion
    %0 = arith.constant 1 : i1
    %e1 = tfhe_rust.create_bool_trivial %bsks, %0 : (!bsks, i1) -> !tfhe_rust.eb
```

(these lines are doing the same thing, so only need one)

**Review by j2kun — APPROVED** [review]
LGTM!

Please ensure your PR is squashed to a single commit before adding the `pull_ready` label.

**Review by asraa — COMMENTED** [review]
Thank you!

**Comment by WoutLegiest** [comment]
Adding Boolean Tfhe-rust types and BitAnd op

**Review by asraa — APPROVED** [review]

**Pull request merged**